### PR TITLE
Refactor msgenc to pass keys via cli arg

### PR DESCRIPTION
**Pull request by PikalaxALT**
Based on discussions in the Discord.

```
msgenc v2021.08.27
Usage: msgenc [-h] [-v] -d|-e [-k KEY] -c CHARMAP INFILE OUTFILE

INFILE        Required: Path to the input file to convert (-e: plaintext; -d: binary).
OUTFILE       Required: Path to the output file (-e: binary; -d: plaintext).
-c CHARMAP    Required: Path to a text file with a character mapping, for example pokeheartgold/charmap.txt.
-d            Decode from binary to text, also print the key
-e            Encode from text to binary using the provided key
-k KEY        The 16-bit encryption key for this message bank. Default: computes it from the binary file name
-v            Print the program version and exit.
-h            Print this message and exit.
```

### CODE REVIEW

File: charmap.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
+// Character mapping for Pokémon HeartGold and SoulSilver
```
Comment (red031000) [inline charmap.txt:1]: thmm yes this is heartgold and soulsilver

**Review by red031000 — CHANGES_REQUESTED** [review]
you can also remove *.key from the .knarcignore

**Comment by PikalaxALT** [comment]
> you can also remove *.key from the .knarcignore

I could, but it's not required to do so.

**Review by red031000 — APPROVED** [review]
ok sure

**Pull request merged**